sphinxlint/checkers.py

```diff
@@ -485,6 +485,38 @@ def check_block(block_lineno, block):
     yield from errors
 
 
+_find_leading_spaces = re.compile(r'^\s*').match
+# finds lines that start with bullets, numbers, and directives' ".."
+_find_list_starters = re.compile(r'^\s*(?:[-+*•‣⁃]|\d+[).]|\(\d+\)|#\.|\.\.)\s+'
+                                 r'(?!index)').match
```

Comment:
This currently combines unnumbered and numbered lists with directives, however we might have to handle them differently, since there are more restrictions for things nested under lists than under directives.